src/gt4py/next/embedded/common.py

```diff
@@ -146,3 +146,31 @@ def _find_index_of_dim(
         if dim == d:
             return i
     return None
+
+
+def canonicalize_any_index_sequence(
+    index: common.AnyIndexSpec,
+) -> common.AnyIndexSpec:
+    new_index: common.AnyIndexSpec = (index,) if isinstance(index, slice) else index
+    if isinstance(index, slice):
+        new_index = _named_slice_to_named_range(index)
```

Comment:
```suggestion
    if isinstance(index, slice):
        new_index = _named_slice_to_named_range(index)
```
after the last change, this is dead code as a slice is now wrapped in a tuple/